pymilvus/client/grpc_handler.py

```diff
@@ -743,91 +724,33 @@ def _check():
 
     @retry_on_rpc_failure()
     def wait_for_loading_partitions(self, collection_name, partition_names, timeout=None):
-        return self._wait_for_loading_partitions(collection_name, partition_names, timeout)
-
-    def _wait_for_loading_partitions(self, collection_name, partition_names, timeout=None):
-        """ Block until load partition complete."""
-
-        request = Prepare.show_partitions_request(collection_name, partition_names, type_in_memory=True)
-
-        while True:
-            future = self._stub.ShowPartitions.future(request, timeout=timeout)
-            response = future.result()
-
-            status = response.status
-            if status.error_code != 0:
-                raise MilvusException(status.error_code, status.reason)
-
-            ol = len(response.partition_names)
-            pl = len(response.inMemory_percentages)
-
-            if ol != pl:
-                raise MilvusException(ErrorCode.UnexpectedError,
-                                      f"len(partition_names) ({ol}) != len(inMemory_percentages) ({pl})")
-
-            loaded_histogram = dict()
-            for i, par_name in enumerate(response.partition_names):
-                loaded_histogram[par_name] = response.inMemory_percentages[i]
-
-            ok = True
-            for par_name in partition_names:
-                if loaded_histogram.get(par_name, 0) != 100:
-                    ok = False
-                    break
+        start = time.time()
 
-            if ok:
-                return
+        def can_loop(t) -> bool:
+            return True if timeout is None else t <= (start + timeout)
 
+        while can_loop(time.time()):
+            progress = self.get_loading_progress(collection_name, partition_names, timeout=timeout)
+            if progress >= 100:
+                break
```

Comment:
the `break` should be `return`, or this function will raise exception when progress >= 100

Reply:
Thanks!